.github/workflows/codecov_ats.yml

```
@@ -80,13 +83,36 @@ jobs:
           env:
             CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
             CODECOV_STATIC_TOKEN: ${{ secrets.CODECOV_STATIC_TOKEN }}
-        # Run label analysis IN DRY MODE (no tests will actually run)
+        # Run label analysis in dry mode to get the list of tests to run
+        # The base commit will be the parent commit (apparently commits on master don't exist in codecov)
         - name: Label Analysis
+          id: label_analysis
           continue-on-error: true
           run: |
-            BASE_COMMIT=$(git merge-base ${{ github.sha }}^ origin/master)
+            set +e
+            BASE_COMMIT=$(git rev-parse ${{ github.sha }}^)
             echo $BASE_COMMIT
-            codecovcli --codecov-yml-path=codecov.yml label-analysis --dry-run --token=${CODECOV_STATIC_TOKEN} --base-sha=${BASE_COMMIT}
+            output=$(codecovcli --codecov-yml-path=codecov.yml label-analysis --dry-run --token=${CODECOV_STATIC_TOKEN} --base-sha=${BASE_COMMIT})
+            if [ $? -eq 0 ]
+            then
+              ATS_TESTS_TO_RUN=$(echo $output | grep -e ATS_TESTS_TO_RUN= | tr -d "ATS_TESTS_TO_RUN=")
+              ATS_TESTS_TO_SKIP=$(echo $output | grep -e ATS_TESTS_TO_SKIP= | tr -d "ATS_TESTS_TO_SKIP=")
+
+              echo "ATS_TESTS_TO_RUN=$ATS_TESTS_TO_RUN" >> "$GITHUB_OUTPUT"
+              echo "ATS_TESTS_TO_SKIP=$ATS_TESTS_TO_SKIP" >> "$GITHUB_OUTPUT"
+
+              test_to_run_count=$(echo $ATS_TESTS_TO_RUN | awk '{print gsub("[ \t]",""); exit}')
+              test_to_skip_count=$(echo $ATS_TESTS_TO_SKIP | awk '{print gsub("[ \t]",""); exit}')
+              echo "Selected $test_to_run_count / $total_tests_count tests to run" >> $GITHUB_STEP_SUMMARY
```

This should be equivalent, but with a bit less edge cases:

```suggestion
              grep -E <<< "$output" 'ATS_TESTS_TO_(RUN|SKIP)=' >> "$GITHUB_OUTPUT"
              source "$GITHUB_OUTPUT"

              testcount() { grep -Eo '[ \t]' | wc -l; }
              run=$(testcount <<< "$ATS_TESTS_TO_RUN")
              skip=$(testcount <<< "$ATS_TESTS_TO_SKIP")
              echo "Selected $run of $((skip + run)) tests to run" >> $GITHUB_STEP_SUMMARY
```

If we had json we could avoid string parsing entirely though:

```
testcount() { jq --arg kind "$1" ".[$kind] | length"; }
run=$(testcount run <<< "$json")
skip=$(testcount skip <<< "$json")
```

FYI: I generally always avoid `awk` because it's a whole scripting language. If that's actually needed, we should write some python.